tests/test_openclhpp.cpp

```diff
@@ -3626,7 +3630,63 @@ void testDevice_GetInfo_CLDeviceName()
     clGetDeviceInfo_StubWithCallback(clGetInfo_testDeviceGetInfoCLDeviceName);
     cl::string deviceName = devicePool[0].getInfo<CL_DEVICE_NAME>();
     TEST_ASSERT_EQUAL_STRING(expected.c_str(), deviceName.c_str());
-} 
+}
+
+#if defined(cl_ext_device_fission)
+static cl_int clCreateSubDevices_testDevice_createSubDevices(
+    cl_device_id in_device, const cl_device_partition_property *properties,
+    cl_uint num_devices, cl_device_id *out_devices, cl_uint *num_devices_ret,
+    int num) {
+    return 0;
+}
+
+static cl_int clCreateSubDevicesEXT_testDevice_createSubDevices(
+    cl_device_id device_in, const cl_device_partition_property_ext *properties,
+    cl_uint n, cl_device_id *out_devices, cl_uint *num, int num_calls) {
+  cl_int ret = CL_SUCCESS;
+
+  TEST_ASSERT_EQUAL(CL_DEVICE_PARTITION_TYPES_EXT, *properties);
+  if(nullptr != out_devices){
+    out_devices[0] = make_device_id(0);
+  }
+  else{
+    *num = 1;
+  }
+  if (device_in == make_device_id(0)) {
+    return CL_SUCCESS;
+  } else if (device_in == make_device_id(1) && nullptr == num) {
+    return CL_DEVICE_NOT_FOUND;
```

Comment:
1. Why does it matter if `nullptr == num`?
2. `CL_DEVICE_NOT_FOUND` is not a valid error code for this function.  Could we pick a valid error code instead, maybe `CL_INVALID_DEVICE` or `CL_INVALID_VALUE`?

Reply:
1. I wanted a function to fail in right place but it can be without this check also
2. Chenged to valid error code